FarsiLibrary.Win/Helpers/EnglishCalendarScroller.cs

```diff
@@ -0,0 +1,123 @@
+﻿using FarsiLibrary.Win.Controls;
+using System;
+
+namespace FarsiLibrary.Win.Helpers
+{
+    class EnglishCalendarScroller : ICalendarScroller
+    {
+        FADatePicker picker;
+
+        private readonly int yearIndex, monthIndex, dayIndex, hourIndex, minuteIndex;
+        private readonly int yearLength, monthLength, dayLength, hourLength, minuteLength;
+        private readonly string datePattern;
+
+        public EnglishCalendarScroller(FADatePicker picker)
+        {
+            this.picker = picker;
+
+            datePattern = picker.mv.MonthViewControl.DefaultCulture.DateTimeFormat.ShortDatePattern;
+
+            yearIndex = datePattern.IndexOf("y");
+            yearLength = datePattern.LastIndexOf("y") - datePattern.IndexOf("y") + 1;
+
+            monthIndex = datePattern.IndexOf("M");
+            monthLength = datePattern.LastIndexOf("M") - datePattern.IndexOf("M") + 1;
+
+
+            dayIndex = datePattern.IndexOf("d");
+            dayLength = datePattern.LastIndexOf("d") - datePattern.IndexOf("d") + 1;
+
+            hourIndex = datePattern.Length + 1;
+            hourLength = 2;
+
+            minuteIndex = hourIndex + 3;
+            minuteLength = 2;
+
+        }
+
+        bool ICalendarScroller.CanScroll
```

Comment:
The other interface methods are explicit, where as this one is implicit. Maybe change them for consistency, e.g.

```suggestion
        public bool CanScroll
```

Reply:
bug fixed